src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -47,14 +54,64 @@ const InfoTile = ({ title, children }) => {
   ])
 }
 
+class EditWorkspaceModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      description: props.workspace.workspace.attributes.description || '',
+      saving: false
+    }
+  }
+
+  async save() {
+    const { onSave, workspace: { workspace: { namespace, name } } } = this.props
+    const { description } = this.state
+    try {
+      this.setState({ saving: true })
+      await Workspaces.workspace(namespace, name).shallowMergeNewAttributes({ description })
+      onSave()
+    } catch (error) {
+      reportError('Error saving workspace', error)
+    } finally {
+      this.setState({ saving: false })
+    }
+  }
+
+  render() {
+    const { onDismiss } = this.props
+    const { description, saving } = this.state
+    return h(Modal, {
+      title: 'Edit workspace',
+      onDismiss,
+      okButton: buttonPrimary({
+        onClick: () => this.save()
+      }, ['Save'])
+    }, [
+      div({ style: styles.label }, ['Description']),
+      h(TextArea, {
+        autoFocus: true,
+        style: { height: 100 },
+        placeholder: 'Enter a description',
+        value: description,
+        onChange: e => this.setState({ description: e.target.value })
+      }),
+      saving && spinnerOverlay
+    ])
+  }
+}
+
 export const WorkspaceDashboard = wrapWorkspace({
   breadcrumbs: () => breadcrumbs.commonPaths.workspaceList(),
   activeTab: 'dashboard'
 },
 class WorkspaceDashboardContent extends Component {
   constructor(props) {
     super(props)
-    this.state = { submissionsCount: undefined, storageCostEstimate: undefined }
+    this.state = {
```

Comment:
Is there any reason to initialize this, since all the values are false-y? If it's just to be explicit about state keys used, I'd rather enforce that in all of our components across the board, rather than selectively.

Reply:
It is to be explicit about state, yes. I'm certainly happy to discuss the idea of enforcing it, though that would be difficult to do automatically, at least with the built-in `setState` mechanisms.

Reply:
Oh, I meant manually, yeah.